### DISCUSSION TITLE
How to add fonts to OnlyOffice container?

### DISCUSSION THREAD

**Discussion by arcstur** [nextcloud/all-in-one#881] [Questions] [answered]
Hello, I would like to install new fonts to my OnlyOffice.

I'm thinking about putting fonts in the server at `/usr/share/fonts`, that would work?

I have the font file.

Thanks in advance for your amazing work!

**Comment by szaimen** [comment]
Hi, we may add an environmental variable that allows to mount a folder from the host into this location.

**Comment by arcstur** [comment]
Hmmm, so currently it's not possible?

**Reply by szaimen** [reply]
Yes

**Reply by arcstur** [reply]
Thanks!!

**Comment by arcstur** [comment]
I was able to add fonts by copying them to the `/usr/share/fonts/custom` folder in the onlyoffice container, but it would have to be done at every restart. I copied fonts to that folder with `docker cp`, I accessed the container with `docker exec -it <container_id> /bin/bash` and ran `/usr/bin/documentserver-generate-allfonts.sh`. At least it's a solution. It would be better if the container just picked up fonts from the machine /usr/share/fonts.

**Comment by szaimen** [comment, marked as answer]
Just discussed this with the management and we will not add this.

**Comment by PeterBenzin** [comment]
Hi, just a noob-kind-of-solution that works for me. I installed the missing fonts locally on my server in the directory:
`/usr/share/fonts/truetype/custom`
The I go to Plesk and open the docker page, where I open the container options and simply added this:
![grafik](https://github.com/nextcloud/all-in-one/assets/36459855/89dd7f35-57c0-4cb6-856b-e9fe1ede4aad)
I hope it survives restarts or updates of the container resp. image.

**Comment by DesertCookie** [comment]
I ended up running a separate container to achieve this. Adding a volume mapping for `/usr/share/fonts/truetype/custom` inside the container allowed me to have a folder on my unRAID host that just holds all the TTF files I want to be able to use. Mainly company fonts and such that are required for branding.

Running a separate container also has the advantage of being able to run an image of your chosing. For me, that's `thomisus/onlyoffice-documentserver-unlimited` that allows for more users and enables mobile editing - though that's really more of a blessing in disguise as OO's mobile apps are absolutely horrible and buggy; but at least editing on tablets now works with this image.

**Comment by webresurs** [comment]
- добавить .ttf фалы из «c:\Windows\Fonts» в «/var/www/nextcloud/apps/documentserver_community/3rdparty/onlyoffice/documentserver/core-fonts» (у вас может быть другой путь) — желательно каждый шрифт в отдельную папку…

- создать .sh файл (отредактируйте путь DIR=".. под ваш)

```
#!/bin/sh


ONLYOFFICE_DATA_CONTAINER=false
if [ "$1" != "" ]; then
    ONLYOFFICE_DATA_CONTAINER=$1
fi

#DIR="/var/www/onlyoffice/documentserver"
DIR="/var/www/nextcloud/apps/documentserver_community/3rdparty/onlyoffice/documentserver"

#Start generate AllFonts.js, font thumbnails and font_selection.bin
echo -n Generating AllFonts.js, please wait...

"$DIR/server/tools/allfontsgen"\
  --input="$DIR/core-fonts"\
  --allfonts-web="$DIR/sdkjs/common/AllFonts.js"\
  --allfonts="$DIR/server/FileConverter/bin/AllFonts.js"\
  --images="$DIR/sdkjs/common/Images"\
  --selection="$DIR/server/FileConverter/bin/font_selection.bin"\
  --output-web="$DIR/fonts"\
  --use-system="true"

chown -R www-data:www-data "$DIR/sdkjs"
chown -R www-data:www-data "$DIR/server/FileConverter/bin"
chown -R www-data:www-data "$DIR/fonts"

echo Done

#Remove gzipped fonts
rm -f $DIR/fonts/*.gz $DIR/sdkjs/common/AllFonts.js.gz
```

- запустить этот скрипт (назвал его GeneratingAllFonts )

`sudo -u www-data bash GeneratingAllFonts.sh`

- перезапустить браузер с предварительной очисткой данных (кэш)
